### DISCUSSION TITLE
How do you merge both code coverages on component and e2e tests

### DISCUSSION THREAD

**Discussion by itstheandre** [cypress-io/cypress#16058] [Component Testing]
I'm sorry if this is repeated, but I can't find my previous post (maybe i didnt send ,also possible. But i wrote it for sure).

Either way, the question i have is:

Is there a way to run the component tests and the e2e tests at the same time in order to have both code-coverage results merged? 

If I do `npx crypress run-ct` it only runs component tests. And if I do `npx cypress run` it only runs integration / e2e tests. Is there a way to have a command that runs both at the same time and have one unified coverage?

Thanks

**Comment by lmiller1990** [comment]
This is not built into Cypress (yet, not sure if/when it will be, since coverage isn't really built in either, but provided via Instanbul.js) but you could try using some of the libraries like [instanbul-combine](https://github.com/jamestalmage/istanbul-combine) or [instanbul-merge](https://github.com/ljharb/istanbul-merge) in the meantime.

It would be nice to combine these eventually, but it's not something that is getting worked on right now. We *are* working on better docs and tooling for component testing code coverage, though, [you can follow the progress here](https://github.com/cypress-io/cypress/issues/16199). I'll probably make a blog post when we have it at a stage where it "just works" and comes built-in to our framework adapters (like `@cypress/react` etc).

**Reply by itstheandre** [reply]
Then, iin order to merge them together, is it possible to, somehow, havee a different "outDir" for each of those code coverags? something like `cypress run-ct --coverage --out ./coverage/comps`

If so it should be fine :)

@lmilleer1990

**Reply by lmiller1990** [reply]
Interesting question... we won't be able to support that exact argument, since it would need to be passed to Instanbul, not Cypress.

I looked into this a bit, it looks like there is a [`report-dir`](https://github.com/istanbuljs/nyc#common-configuration-options) cli option, [and this issue comment](https://github.com/istanbuljs/nyc/issues/63#issuecomment-173472890), but it's not entirely clear how you might do it from the README. It looks like you can configure a ` instanbul.yml` and pass a [`dir`](https://github.com/atlassian/techradar/blob/master/.istanbul.yml#L23) argument... although changing this file dynamically depending on your runner is not a nice solution.

I am curious the best way to accomplish this, I'll keep thinking about it. I think this is something we should be able to do via userland config, or `cypress/code-coverage`, without involving Cypress core too much.

I wonder if @bahmutov has some ideas, he has written about [merging coverage before](https://github.com/bahmutov/cypress-and-jest).
